cpmpy/expressions/core.py

```diff
@@ -140,8 +140,20 @@ def _deepcopy_args(self, memodict={}):
             Used in copy() methods of expressions to ensure there are no shared variables between the original expression and its copy.
             :param: memodict: dictionary with already copied objects, similar to copy.deepcopy()
         """
+        return self._deepcopy_arg_list(self.args)
+
+    def _deepcopy_arg_list(self, arglist, memodict={}):
+        """
+            Create and return a deep copy of the arguments in `arglist`.
+            Recursively deep copy nested lists.
+            :param: memodict: dictionary with already copied objects, similar to copy.deepcopy()
+        """
         copied = []
-        for arg in self.args:
+        for arg in arglist:
+            if is_any_list(arg):
+                copied.append(self._deepcopy_arg_list(arg, memodict))
```

Comment:
In most places in the code we use `lst += [new_element]` instead of `lst.append(new_element)`. But this is fine too :)

Reply:
Okay, changed it, it's better to be consistent.